ethapi/block_cert_json_test.go

```diff
@@ -90,3 +92,93 @@ func TestBlockCertificateToJson(t *testing.T) {
 	require.Equal(bitset, json.Signers)
 	require.Equal(sig, json.Signature)
 }
+
+func TestBlockCertificate_JsonEncodingMatchesExpectedFormat(t *testing.T) {
+	tests := map[string]cert.BlockCertificate{
+		"empty":     makeEmptyBlockCert(),
+		"non-empty": makeTestBlockCert(t),
+	}
+
+	for name, cert := range tests {
+		t.Run(name, func(t *testing.T) {
+			validateBlockCertJsonFormat(t, cert)
+		})
+	}
+}
+
+func validateBlockCertJsonFormat(t *testing.T, cert cert.BlockCertificate) {
+	hashRegex := `"0x[0-9a-f]{64}"`
+	tests := map[string]string{
+		"chainId":   `"chainId":\d+`,
+		"number":    `"number":\d+`,
+		"hash":      `"hash":` + hashRegex,
+		"stateRoot": `"stateRoot":` + hashRegex,
+		"signers":   `"signers":("0x[0-9a-f]+"|null)`,
+		"signature": `"signature":"0x[0-9a-f]{192}"`,
+	}
+
+	for name, regex := range tests {
+		t.Run(name, func(t *testing.T) {
+			require := require.New(t)
+			data, err := json.Marshal(toJsonBlockCertificate(cert))
+			require.NoError(err)
+			require.Regexp(regexp.MustCompile(regex), string(data))
+		})
+	}
+}
+
+func TestBlockCertificate_ContainsExpectedValues_EmptyCertificate(t *testing.T) {
+	require := require.New(t)
+	emptyCert := makeEmptyBlockCert()
+	data, err := json.Marshal(toJsonBlockCertificate(emptyCert))
+	require.NoError(err)
+	require.Contains(string(data), `"chainId":0`)
+	require.Contains(string(data), `"number":0`)
+	require.Contains(string(data), `"hash":"0x`)
```

Comment:
nit: this is a bit misleading ... it actually contains a `"hash":"0x0000000000000000000000000000000000000000000000000000000000000000"`, but the test, at a first glance, makes it look like a `"hash":"0x"` is expected -- the missing trailing `"` is missing, which is very subtile. The same for the state root and signature below.

A readable alternative may be something like
```suggestion
	require.Contains(string(data), `"hash":"0x` + strings.Repeat("00", 32) + `"`)
```